oap-stdlib/src/main/java/oap/application/Kernel.java

```diff
@@ -295,7 +301,8 @@ private void linkLinks( ServiceInitialization initialization ) {
 
                         var linkMethod = reflect.method( "add" + methodSuffix ).orElse( null );
                         if( linkMethod == null ) linkMethod = reflect.method( "set" + methodSuffix ).orElse( null );
-                        if( linkMethod == null ) linkMethod = reflect.method( "add" + methodSuffix + "Listener" ).orElse( null );
+                        if( linkMethod == null )
```

Comment:
it is easy to make the mistake here in if-else statements now or later in the process of refactoring. 
Please add curly braces, they also will help to understand the code better.